internal/layer2/announcer.go

```diff
@@ -181,6 +225,8 @@ func (a *Announce) shouldAnnounce(ip net.IP) dropReason {
 
 // SetBalancer adds ip to the set of announced addresses.
 func (a *Announce) SetBalancer(name string, ip net.IP) {
+	// Call spamStart at the end of the function without holding the lock
+	defer a.spamStart(name)
```

Comment:
It looks like these changes make it so we're sending GARP for each service using the same IP, bypassing the logic that used to skip it when we detected it was the same IP in use by more than one service.  Or if that's handled, can you help with what I missed?

Reply:
Good catch, indeed I spam once per name and not per IP, I'll change that

Reply:
Done, it even simplified a bit the code in the end